state/cloudcredentials.go

```diff
@@ -249,3 +277,32 @@ func (st *State) WatchCredential(cred names.CloudCredentialTag) NotifyWatcher {
 	}
 	return newNotifyCollWatcher(st, cloudCredentialsC, filter)
 }
+
+func cloudCredentialIncRefOp(st modelBackend, tag names.CloudCredentialTag) (txn.Op, error) {
+	refcounts, closer := st.db().GetCollection(globalrefcountsC)
+	defer closer()
+
+	refKey := cloudCredentialGlobalKey(tag)
+	op, err := nsRefcounts.CreateOrIncRefOp(refcounts, refKey, 1)
+	return op, errors.Annotate(err, "increment cloudcredentials reference")
+}
+
+func cloudCredentialDecRefOps(st modelBackend, tag names.CloudCredentialTag) ([]txn.Op, error) {
+	refcounts, closer := st.db().GetCollection(globalrefcountsC)
+	defer closer()
+
+	refKey := cloudCredentialGlobalKey(tag)
+	assertOp, n, err := nsRefcounts.CurrentOp(refcounts, refKey)
```

Comment:
I'm not convinced this is a good idea. We shouldn't be trying to decref more than necessary, so I think doing an unconditional AliveDecRefOp is most appropriate. Checking first just means we're allowing for double decrefs doesn't it? And if something else increments in the mean time, a double decref will put things into a bad state.